src/ralph/virtual/admin.py

```diff
@@ -52,11 +54,30 @@ class VirtualServerTypeForm(RalphAdmin):
 
 
 class VirtualServerForm(RalphAdminForm):
+    HYPERVISOR_TYPE_ERR_MSG = 'Hypervisor must be one of \
+                                   DataCenterAsset or VirtualServer.'
+    HYPERVISOR_IS_SELF_ERR_MSG = 'Hypervisor and host must be different.'
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         if 'parent' in self.fields:
             self.fields['parent'].required = True
 
+    def is_valid(self):
+        valid = super().is_valid()
+        virtual_server = ContentType.objects.get_for_model(VirtualServer)
+        dc_asset = ContentType.objects.get_for_model(DataCenterAsset)
+        selected_object = self.cleaned_data['parent']
+        if selected_object.content_type not in (virtual_server, dc_asset):
```

Comment:
why do we allow to VM be parent of another VM?

Reply:
Well, I have changed that to CloudHost. As you have said, `DataCenterAssets` are added as cloud hosts for Ironic.